ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndexSpec.scala

```diff
@@ -115,6 +116,101 @@ final class LedgerConfigurationSubscriptionFromIndexSpec
         }
     }
 
+    "not use the configuration from a rejection" in {
+      val configurationEntries = List(
+        offset("0123") -> ConfigurationEntry.Rejected(
+          submissionId = "submission ID",
+          rejectionReason = "rejected because we felt like it",
+          proposedConfiguration = Configuration(
+            generation = 10,
+            timeModel = LedgerTimeModel.reasonableDefault.copy(maxSkew = Duration.ZERO),
+            maxDeduplicationTime = Duration.ZERO,
+          ),
+        )
+      )
+      val configurationLoadTimeout = 5.seconds
+
+      val index = mock[IndexConfigManagementService]
+      when(index.lookupConfiguration()).thenReturn(Future.successful(None))
+      when(index.configurationEntries(None))
+        .thenReturn(Source(configurationEntries).concat(Source.never))
+      val scheduler = new ExplicitlyTriggeredScheduler(null, NoLogging, null)
+      val subscriptionBuilder = new LedgerConfigurationSubscriptionFromIndex(
+        indexService = index,
+        scheduler = scheduler,
+        materializer = materializer,
+        servicesExecutionContext = system.dispatcher,
+      )
+
+      subscriptionBuilder
+        .subscription(configurationLoadTimeout)
+        .use { currentLedgerConfiguration =>
+          Delayed.by(100.millis)(()).map { _ =>
+            currentLedgerConfiguration.latestConfiguration() should be(None)
```

Comment:
```suggestion
            currentLedgerConfiguration.latestConfiguration should be(None)
```

Reply:
Nope, this is definitely a method that depends on side effects, and therefore should not be confused with a stateless property.